ports/rp2/rp2_flash.c

```diff
@@ -122,7 +128,13 @@ STATIC mp_obj_t rp2_flash_ioctl(mp_obj_t self_in, mp_obj_t cmd_in, mp_obj_t arg_
             return MP_OBJ_NEW_SMALL_INT(BLOCK_SIZE_BYTES);
         case MP_BLOCKDEV_IOCTL_BLOCK_ERASE: {
             uint32_t offset = mp_obj_get_int(arg_in) * BLOCK_SIZE_BYTES;
+            #if MICROPY_HW_USB_MSC
+            uint32_t ints = save_and_disable_interrupts();
```

Comment:
Why are the interrupts disabled here? What do you want to protect from?

Reply:
To prevent contention with the host/USB over the flash.

Reply:
But what you mean by that? You mean that by disabling interrupts we make sure that the USB host will not be able to access the flash as those accesses are done from the interrupt handlers?

Reply:
Yes exactly.